device/celestica/x86_64-cel_seastone-r0/sonic_platform/psu.py

```diff
@@ -219,5 +251,148 @@ def get_status(self):
             A boolean value, True if device is operating properly, False if not
         """
         raw = self.__get_gpio_value(
-            self.dx010_psu_gpio[self.index+1]['status'])
+            self.dx010_psu_gpio[self.index + 1]['status'])
         return int(raw, 10) == 1
+
+    def get_model(self):
+        """
+        Retrieves the model number (or part number) of the device
+        Returns:
+            string: Model/part number of device
+        """
+        if not os.path.exists(self.eeprom_addr):
+            os.system('echo 24c02 0x{} > /sys/bus/i2c/devices/i2c-{}/new_device'.format(PSU_I2C_MAPPING[self.index]["eeprom_addr"], self.i2c_num))
+            time.sleep(5)
+        model = self.read_fru(self.eeprom_addr, TLV_ATTR_TYPE_MODEL)
+        if not model:
+            return NotImplementedError
+        return model
+
+    def get_serial(self):
+        """
+        Retrieves the serial number of the device
+        Returns:
+            string: Serial number of device
+        """
+        if not os.path.exists(self.eeprom_addr):
+            os.system('echo 24c02 0x{} > /sys/bus/i2c/devices/i2c-{}/new_device'.format(PSU_I2C_MAPPING[self.index]["eeprom_addr"], self.i2c_num))
+            time.sleep(5)
+        serial = self.read_fru(self.eeprom_addr, TLV_ATTR_TYPE_SERIAL)
+        if not serial:
+            return NotImplementedError
```

Comment:
Can we set the default return to 'N/A' to prevent errors?

Reply:
ok